library/core/src/mem/maybe_uninit.rs

```diff
@@ -913,7 +913,11 @@ impl<T> MaybeUninit<T> {
     /// };
     /// ```
     #[stable(feature = "maybe_uninit_ref", since = "1.55.0")]
-    #[rustc_const_unstable(feature = "const_maybe_uninit_assume_init", issue = "none")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
+    #[rustc_const_stable(
+        feature = "const_maybe_uninit_assume_init",
```

Comment:
```suggestion
        feature = "const_maybe_uninit_assume_init_mut",
```
Do we care about the feature gate? All the other `assume_init` get quite specific const feature gates so this only seems consistent... but it doesn't really matter any more once it's stable anyway. Seems a bit silly that the `const_maybe_uninit_assume_init` feature gate does not contain the `assume_init` function.

Reply:
The reason that this is the case is because every other feature got incrementally carved out of `const_maybe_uninit_assume_init`, and people okayed it instead of pausing to reconsider how the feature was divided. By renaming and then stabilizing a feature name that didn't previously exist, we make the history notably harder to understand, because it's now a feature that never had an issue open for it. So no. We should not.

Reply:
IMO we make the history less confusing by picking feature names that reflect what is actually covered by the feature. :shrug: 


it is also unclear to me in which sense this makes history "harder to understand" -- we remove the tracking issue number upon stabilization, so clearly there's no expectation of tracking history that way. And `git blame` will work just fine either way.

If you are thinking of people searching for the feature name in the issue tracker, it would be trivial to edit the tracking issue to also mention the split-off feature gates, and if that is something people care about it should have been done as part of the previous PR that stabilized parts of this feature. Furthermore, to my knowledge we don't have a standard or expectation about feature names along the lines of what you are saying here, so it seems futile to try to partially enforce it.

It seems to make you're making up a new standard here and now trying to unilaterally enforce it.

Reply:
Is
this
patch
required?

Reply:
Or is it merely an aesthetic preference?

Reply:
"required" is a strong term. That's ultimately for libs-api to decide.

I just made my case for why I personally think his PR is better with the patch than without it.

Reply:
Yes, I'm the kind of idiot who tries to search the issue tracker for feature names and is bothered when these renames happen frivolously. I understand some renames must happen, but I dislike it when they do. We usually stabilize things under their existing feature name, often even if that feature name doesn't make sense, unless there has to be a real split, or unless the split is premeditated.

You may laugh at me if you like for being such a fool as to believe we give features these names for a reason beyond simply whatever must be plugged in to Rust source to enable the feature.